snuba/consumers/consumer_builder.py

```diff
@@ -83,62 +83,98 @@ def __init__(
         mock_parameters: Optional[MockParameters] = None,
         cooperative_rebalancing: bool = False,
     ) -> None:
-        self.storage = get_writable_storage(storage_key)
+        self.storages = {key: get_writable_storage(key) for key in storage_keys}
         self.bootstrap_servers = kafka_params.bootstrap_servers
         self.consumer_group = kafka_params.group_id
-        topic = (
-            self.storage.get_table_writer()
+
+        # in multistorage case, we ensure there is one topic in common
+        topics = {
+            storage.get_table_writer()
             .get_stream_loader()
             .get_default_topic_spec()
             .topic
-        )
+            for storage in self.storages.values()
+        }
+        topic = topics.pop()
+        if topics:
+            raise ValueError("only one topic is supported")
+
+        if len(self.storages) == 1:
```

Comment:
I think the goal would be to ideally share all of this implementation, not move a ton of if/else statements into this function